ui/src/components/infinite-scroll/QasInfiniteScroll.vue

```diff
@@ -0,0 +1,161 @@
+  </div>
+</template>
+
+<script setup>
+import { ref, computed, defineExpose, inject } from 'vue'
+import { NotifyError } from '../../plugins'
+
+defineOptions({ name: 'QasInfiniteScroll' })
+
+const props = defineProps({
+  list: {
+    type: Array,
+    default: () => []
+  },
+
+  limitPerPage: {
+    type: Number,
+    default: 12
+  },
+
+  url: {
+    type: String,
+    default: '',
+    required: true
+  },
+
+  params: {
+    type: Object,
+    default: () => ({})
+  },
+
+  infiniteScrollProps: {
+    type: Object,
+    default: () => ({})
+  },
+
+  maxHeight: {
+    type: String,
+    default: ''
+  }
+})
+
+defineExpose({ refreshList, removeItemByIndex })
+
+const emits = defineEmits(['update:list'])
+
+const axios = inject('axios')
+
+const hasFetchingError = ref(false)
+const isFetching = ref(false)
+const hasMadeFirstFetch = ref(false)
+const infiniteScroll = ref(null)
+const count = ref(0)
+const offset = ref(0)
+
+const listLength = computed(() => model.value.length)
+
+const attributes = computed(() => ({
+  offset: 100,
+  debounce: 0,
+  ...(props.maxHeight && { scrollTarget: '.qas-infinite-scroll' }),
+  ...props.infiniteScrollProps
+}))
+
+const isEmptyList = computed(() => !listLength.value && !isFetching.value)
+
+const hasNoResults = computed(() => isEmptyList.value && hasMadeFirstFetch.value)
+
+const containerStyle = computed(() => ({
+  ...(props.maxHeight && { maxHeight: props.maxHeight, overflow: 'auto' })
+}))
+
+const model = computed({
+  get () {
+    return props.list
+  },
+
+  set (newList) {
+    emits('update:list', newList)
+  }
+})
+
+async function onLoad (_, done) {
+  const hasMadeFirstFetchAndHasNoData = hasMadeFirstFetch.value && !listLength.value
+  const hasFetchAllData = listLength.value && listLength.value >= count.value
+  const stop = hasFetchingError.value || isFetching.value || hasMadeFirstFetchAndHasNoData || hasFetchAllData
+
+  if (stop) {
+    infiniteScroll.value.stop()
+  } else {
+    infiniteScroll.value.resume()
+    await fetchList()
+  }
+
+  done()
+}
+
+async function fetchList () {
+  try {
+    isFetching.value = true
+
+    const { data } = await axios.get(props.url, {
+      params: { offset: offset.value, limit: props.limitPerPage, ...props.params }
+    })
+
+    const newList = [...model.value, ...(data.results || [])]
+
+    model.value = newList
+    offset.value = newList.length
+    count.value = data.count
+
+    // Sinalizar que houve já uma busca, para evitar que onLoad entre em looping, após buscar uma vez e retornar uma lista vazia.
+    hasMadeFirstFetch.value = true
+  } catch {
+    NotifyError('Ops… Não conseguimos acessar as informações. Por favor, tente novamente em alguns minutos.')
+
+    hasFetchingError.value = true
+  } finally {
+    isFetching.value = false
+  }
+}
+
+function refreshList () {
+  count.value = 0
+  offset.value = 0
+  model.value = []
+
+  hasMadeFirstFetch.value = false
+
+  this.$nextTick(() => {
```

Comment:
Cara, tu testou isto? Este `this.$nextTick` não vai funcionar aqui, tu precisa importar o `nextTickt` do vue.

Reply:
ajustado